unified-runtime/source/adapters/level_zero/image.cpp

```diff
@@ -460,499 +125,15 @@ ur_result_t urBindlessImagesImageCopyExp(
-      auto *UrImage = static_cast<const _ur_image *>(pSrc);
-      char *DstPtr =
-          static_cast<char *>(pDst) + pCopyRegion->dstOffset.z * DstSlicePitch +
-          pCopyRegion->dstOffset.y * DstRowPitch +
-          pCopyRegion->dstOffset.x * getPixelSizeBytes(pDstImageFormat);
-      ZE2UR_CALL(zeCommandListAppendImageCopyToMemoryExt,
-                 (ZeCommandList, DstPtr, UrImage->ZeImage, &SrcRegion,
-                  DstRowPitch, DstSlicePitch, ZeEvent, WaitList.Length,
-                  WaitList.ZeEventList));
-    } else {
-      // Copy from pitched USM memory to host
-      ze_copy_region_t ZeDstRegion = {(uint32_t)pCopyRegion->dstOffset.x,
-                                      (uint32_t)pCopyRegion->dstOffset.y,
-                                      (uint32_t)pCopyRegion->dstOffset.z,
-                                      DstRowPitch,
-                                      (uint32_t)pCopyRegion->copyExtent.height,
-                                      (uint32_t)pCopyRegion->copyExtent.depth};
-      uint32_t SrcRowPitch = pSrcImageDesc->rowPitch;
-      ze_copy_region_t ZeSrcRegion = {(uint32_t)pCopyRegion->srcOffset.x,
-                                      (uint32_t)pCopyRegion->srcOffset.y,
-                                      (uint32_t)pCopyRegion->srcOffset.z,
-                                      SrcRowPitch,
-                                      (uint32_t)pCopyRegion->copyExtent.height,
-                                      (uint32_t)pCopyRegion->copyExtent.depth};
-      uint32_t SrcSlicePitch = 0;
-      ZE2UR_CALL(zeCommandListAppendMemoryCopyRegion,
-                 (ZeCommandList, pDst, &ZeDstRegion, DstRowPitch, DstSlicePitch,
-                  pSrc, &ZeSrcRegion, SrcRowPitch, SrcSlicePitch, ZeEvent,
-                  WaitList.Length, WaitList.ZeEventList));
-    }
-  } else if (imageCopyFlags == UR_EXP_IMAGE_COPY_FLAG_DEVICE_TO_DEVICE) {
-    ze_image_region_t DstRegion;
-    UR_CALL(getImageRegionHelper(ZeImageDesc, &pCopyRegion->dstOffset,
-                                 &pCopyRegion->copyExtent, DstRegion));
-    ze_image_region_t SrcRegion;
-    UR_CALL(getImageRegionHelper(ZeImageDesc, &pCopyRegion->srcOffset,
-                                 &pCopyRegion->copyExtent, SrcRegion));
-    auto *UrImageDst = static_cast<_ur_image *>(pDst);
-    auto *UrImageSrc = static_cast<const _ur_image *>(pSrc);
-    ZE2UR_CALL(zeCommandListAppendImageCopyRegion,
-               (ZeCommandList, UrImageDst->ZeImage, UrImageSrc->ZeImage,
-                &DstRegion, &SrcRegion, ZeEvent, WaitList.Length,
-                WaitList.ZeEventList));
-  } else {
-    logger::error("urBindlessImagesImageCopyExp: unexpected imageCopyFlags");
-    return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
-  }
-
-  UR_CALL(hQueue->executeCommandList(CommandList, Blocking, OkToBatch));
-
-  return UR_RESULT_SUCCESS;
-}
+  auto res = bindlessImagesHandleCopyFlags(
```

Comment:
I think the naming here could be improved. Inside this function it is processing the copy requested, calling the approperate UR copy functions. Maybe `bindlessImagesProcessCopy` is better?

What do you think?